src/kernel/multitasking/tasks/task_small.c

```diff
@@ -39,136 +46,197 @@ static task_small_t* _tasking_get_next_task(task_small_t* previous_task) {
     return next_task;
 }
 
+static task_small_t* _tasking_get_next_runnable_task(task_small_t* previous_task) {
+    task_small_t* iter = _tasking_get_next_task(previous_task);
+    for (int i = 0; i < MAX_TASKS; i++) {
+        if (iter->blocked_info.status != RUNNABLE) {
+            iter = _tasking_get_next_task(iter);
+            continue;
+        }
+        return iter;
+    }
+    panic("couldn't find runnable task");
+}
+
```

Comment:
Simple round-robin scheduler. The MLFQ scheduler should be reenabled in the future once the responsiveness of the system would benefit from it.